packages/appium/docs/zh/ecosystem/build-drivers.md

```diff
@@ -1,5 +1,758 @@
+已经有许多开源 Appium 驱动程序, 您可以查看以供参考. 
+有一个 [假驱动程序](https://github.com/appium/appium/tree/2.0/packages/fake-driver) 示例驱动程序, 
+其中除了展示本指南中描述的一些内容外, 基本上什么都不做.
+
+当然, Appium 的所有官方驱动程序都是开源的, 
+可在该项目的 GitHub 组织的仓库中找到. 
+因此, 如果你发现自己问, "司机如何做X?",
+阅读这些驱动程序的代码!
+如果你被卡住了, 也不要害怕向Appium开发人员提问;
+我们总是很乐意帮助确保驱动程序开发体验良好!
+
+## Appium 驱动程序的基本要求
+
+这些是您的驱动程序*必须*执行 (或成为) 的操作, 如果您希望它成为有效的Appium驱动程序.
+
+### 带有 Appium 扩展元数据的 Node.js 包
+
+所有 Appium 驱动程序基本上都是 Node.js 软件包, 
+因此必须具有有效的`package.json`. 
+您的驱动程序不 _限于_  Node.js, 
+但它必须提供用 Node.js 编写的适配器以便 Appium 可以加载它. 
+
+您的 `package.json` 必须包含 `appium` 作为 `peerDependency`. 
+对依赖项版本应尽可能宽松 (除非您碰巧知道您的驱动程序只会
+适用于某些版本的 Appium). 例如, 针对Appium 2.0, 大致上是
+`^2.0.0`, 声明您的驱动程序适用于以 2.x 开头的任何版本的 Appium.
+
+您的 `package.json` 必须包含一个 `appium` 字段, 
+如下所示(我们称之为 'Appium扩展元数据'):
+
+    ```json
+    {
+      ...,
+      "appium": {
+        "driverName": "fake",
+        "automationName": "Fake",
+        "platformNames": [
+          "Fake"
+        ],
+        "mainClass": "FakeDriver"
+      },
+      ...
+    }
+    ```
+
+必填子字段为:
+
+* `driverName`: 这应该是驱动程序的短名称.
+* `automationName`: 这应该是用户告诉 Appium 使用*您的*驱动程序用于其 `appium:automationName`功能的字符串.
+* `platformNames`: 这是一个或多个平台名称的数组, 这些名称被认为对驱动程序有效. 
+  当用户发送 `platformName` 功能以启动会话时, 它必须包含在
+  此列表供驱动程序处理会话. 已知的平台名称字符串包括:`iOS`,
+  `tvOS`, `macOS`, `Windows`, `Android`.
+* `mainClass`: 这是来自 `main` 字段的命名导出(采用 CommonJS 样式). 它必须是
+  扩展 Appium 的 `BaseDriver` 的类 (见下文).
+
+### 扩展 Appium 的 `BaseDriver` 类
+
+最终, 您的驱动程序会更容易编写, 
+因为实现WebDriver 协议和处理某些常见逻辑已经由 Appium 负责. 
+所有这些都被编码为 Appium 导出类以供您使用, 称为 `BaseDriver`. 
+其被从 `appium/driver`中导出, 
+因此您可以使用这些样式之一来导入它并创建 *自己的* 类以扩展:
+
+```js
+import {BaseDriver} from 'appium/driver';
+// or: const {BaseDriver} = require('appium/driver');
+
+export class MyDriver extends BaseDriver {
+}
+```
+
+### 使您的驱动程序可用
+
+基本上就是这样!使用 Node.js 包导出驱动程序类并使用正确的 Appium
+扩展元数据, 您已经有了 Appium 驱动程序!
+现在它不 *做* 任何事情, 但你可以在 Appium 中加载它, 
+使用它启动和停止会话, 等等...
+
+要使其可供用户使用, 您可以通过 NPM 发布它. 
+当您这样做时, 您的驱动程序将可通过 Appium CLI 安装:
+
+```
+appium driver install --source=npm <driver-package-on-npm>
+```
+
+当然, 最好先测试您的驱动程序. 
+了解其在 Appium 中如何工作的一种方法是先在本地安装:
+
+```
+appium driver install --source=local /path/to/your/driver
+```
+
+### 开发驱动程序
+
+如何开发驱动程序取决于您. 
+但是, 从Appium内部运行很方便, 无需进行大量发布和安装. 
+最直接的方法是将最新版本的 Appium 作为 `devDependency`包含在内, 
+然后再包括您自己的驱动程序, 例如这样:
+
+```json
+{
+    "devDependencies": {
+        ...,
+        "appium": "^2.0.0",
+        "your-driver": "file:.",
+        ...
+    }
+}
+```
+
+现在, 您可以在本地运行 Appium (`npm exec appium` 或 `npx appium`),
+并且因为您的驱动程序是作为依赖项与它一起列出, 它将自动 "安装" 并可用. 
+您可以以这种方式设计您的 E2E 测试, 或者若您在 Node.js 中编写它们, 
+您可以简单地导入Appium 的启动服务器方法, 用于处理 Node 中 Appium 服务器的启动和停止. 
+(TODO: 准备好后, 在其中一个开源驱动程序中引用此实现).
+
+使用现有 Appium 服务器进行本地开发的另一种方法是, 
+简化安装您的本地驱动程序:
+
+```
+appium driver install --source=local /path/to/your/driver/dev/dir
+```
+
+## 标准驱动程序实现理念
+
+这些是创建驱动程序时可能会发现自己想要做的事情.
+
+### 在构造函数中设置状态
+
+如果你定义了自己的构造函数, 
+你需要调用 `super` 来确保所有标准状态设置正确:
+
+```js
+constructor(args) {
```

Comment:
probably want to pull in the fixes from my recent PR, should be like `...args` in both places instead

Reply:
Updated~